src/auth/oauth2client.ts

```diff
@@ -500,21 +503,26 @@ export class OAuth2Client extends AuthClient {
                 null, response);
             return;
           }
-          const cacheControl = response.headers['cache-control'];
+
+          let cacheControl = response ? response.headers['cache-control'] : '';
+          if (cacheControl instanceof Array) {
+            cacheControl = cacheControl.length > 0 ? cacheControl[0] : '';
+          }
+
           let cacheAge = -1;
           if (cacheControl) {
             const pattern = new RegExp('max-age=([0-9]*)');
             const regexResult = pattern.exec(cacheControl);
-            if (regexResult.length === 2) {
+            if (regexResult && regexResult.length === 2) {
```

Comment:
Not sure if you'd prefer this, but I find the following more readable for regexp.exec:

```js
const [match, maxAge] = pattern.exec(cacheControl) || [];
if (match) cacheAge = maxAge;
```